tasks.py

```diff
@@ -138,16 +101,63 @@ def codebuild(ctx, revision):
             break
 
 
+@task(pre=[production_failsafe])
+def stack_create(ctx):
+    """
+    Package & upload the lambda function code and build the CloudFormation stack
+    """
+    for func in names.FUNCTIONS:
+        __build_function(ctx, func, upload_to_s3=True)
+
+    ctx.run(f"cdk deploy -c VIA_INVOKE=true {profile_arg()}")
+
+
+@task(pre=[production_failsafe])
+def stack_update(ctx):
+    """
+    Updates the CloudFormation stack (use the deploy.* tasks to update functions)
+    """
+    ctx.run(f"cdk deploy -c VIA_INVOKE=true {profile_arg()}")
+
+
+@task
+def stack_diff(ctx):
+    """
+    Output a cdk diff of the Cloudformation stack
+    """
+    ctx.run(f"cdk -c VIA_INVOKE=true diff {profile_arg()}")
+
+
+@task
+def stack_synth(ctx):
+    """
+    Output the cdk-generated CloudFormation template
+    """
+    ctx.run(f"cdk synth -c VIA_INVOKE=true {profile_arg()}")
+
+
+@task
+def stack_list(ctx):
+    """
+    Outputs the name of the cdk CloudFormation stack
+    """
+    ctx.run(f"cdk list -c VIA_INVOKE=true {profile_arg()}")
+
+
+@task
+def stack_delete(ctx):
+    """
+    Deletes the cdk CloudFormation stack
+    """
+    ctx.run(f"cdk destroy -c VIA_INVOKE=true {profile_arg()}")
```

Comment:
this needs to be preceded by emptying the bucket. Even with the retention policy set to `DESTROY` cloudformation won't delete a non-empty bucket: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket.html